### PR TITLE
Changes to make tagging file tests data driven #2937

### PR DESCRIPTION
**Pull request by joachimmetz**
Changes to make tagging file tests data driven #2937

### CODE REVIEW

File: tests/data/tag_linux.py [hunk 1]
```diff
@@ -32,84 +32,114 @@ class LinuxTaggingFileTest(test_lib.TaggingFileTestCase):
 
   _TEST_TIMESTAMP = timelib.Timestamp.CopyFromString('2020-04-04 14:56:39')
 
+  def _CheckTaggingRule(
+      self, event_data_class, attribute_values_per_name, expected_rule_names):
+    """Tests a tagging rule.
+
+    Args:
+      event_data_class (type): class of the even data object to use in tests.
```
Comment (Onager) [inline tests/data/tag_linux.py:40]: typo, event not even

File: tests/data/tag_linux.py [hunk 2]
```diff
@@ -32,84 +32,114 @@ class LinuxTaggingFileTest(test_lib.TaggingFileTestCase):
...
+      attribute_values_per_name (dict[str, list[str]): attribute values per
```
Comment (Onager) [inline tests/data/tag_linux.py:41]: Please clarify the docstring here - I don't understand what's supposed to be in this dict.

File: tests/data/tag_linux.py (hunk 1 above)
Comment (joachimmetz) [reply tests/data/tag_linux.py:40]: ack

File: tests/data/tag_linux.py (hunk 2 above)
Comment (joachimmetz) [reply tests/data/tag_linux.py]: updated

File: tests/data/tag_linux.py [hunk 3]
```diff
@@ -32,84 +32,115 @@ class LinuxTaggingFileTest(test_lib.TaggingFileTestCase):
 
   _TEST_TIMESTAMP = timelib.Timestamp.CopyFromString('2020-04-04 14:56:39')
 
+  def _CheckTaggingRule(
+      self, event_data_class, attribute_values_per_name, expected_rule_names):
+    """Tests a tagging rule.
+
+    Args:
+      event_data_class (type): class of the event data object to use in tests.
+      attribute_values_per_name (dict[str, list[str]): values of the event data
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply tests/data/tag_linux.py:41]: yes it is that, please provide a suggestion that makes more sense to you or let's chat about this to not have to do many round trips via github.

**Comment by joachimmetz** [comment]
Moved the `_CheckTaggingRule` method to testlib in anticipation of changing the other test files and waiting for this review to be completed.

**Pull request merged**